sbtc-core/src/signer/config.rs

```diff
@@ -1,16 +1,245 @@
-use bitcoin::Address as BitcoinAddress;
-use secp256k1::PublicKey;
+use bitcoin::{Address as BitcoinAddress, Network};
+use serde::Deserialize;
 use stacks_core::address::StacksAddress;
+use std::str::FromStr;
+use toml;
+
+use crate::{SBTCError, SBTCResult};
 
 #[derive(Clone, Debug)]
 /// Configuration for the signer approval/denial
 pub struct Config {
-    /// The maximum dollar amount of a transaction that will be auto approved
-    pub auto_approve_max_amount: u64,
-    /// The public key of the signer being delegated to
-    pub delegate_public_key: PublicKey,
+    /// The maximum dollar amount of a transaction that will be auto approved.
+    pub auto_approve_max_amount: Option<u64>,
     /// The BTC addresses to be auto denied
     pub auto_deny_addresses_btc: Vec<BitcoinAddress>,
     /// The STX addresses to be auto denied
     pub auto_deny_addresses_stx: Vec<StacksAddress>,
+    /// The number of blocks before deadline at which point the transaction will be auto denied. Default is 10 blocks.
+    pub auto_deny_deadline_blocks: u32,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            auto_approve_max_amount: None,
+            auto_deny_addresses_btc: vec![],
+            auto_deny_addresses_stx: vec![],
+            auto_deny_deadline_blocks: 10,
+        }
+    }
+}
+
+impl Config {
+    /// Load the configuration from a toml file
+    pub fn from_path(path: impl AsRef<std::path::Path>) -> SBTCResult<Self> {
+        let config_toml = ConfigTOML::from_path(path)?;
+        Self::try_from(config_toml)
+    }
+}
+
+impl TryFrom<ConfigTOML> for Config {
+    type Error = SBTCError;
+    fn try_from(config: ConfigTOML) -> SBTCResult<Self> {
+        let auto_approve_max_amount = config.auto_approve_max_amount;
+        let mut auto_deny_addresses_btc = Vec::new();
+        let mut auto_deny_addresses_stx = Vec::new();
+        for address in config.auto_deny_addresses_btc.unwrap_or_default() {
+            auto_deny_addresses_btc.push(BitcoinAddress::from_str(&address).map_err(|e| {
+                SBTCError::InvalidConfig(format!("Failed to parse bitcoin address: {}", e))
+            })?);
+        }
+        for address in config.auto_deny_addresses_stx.unwrap_or_default() {
+            auto_deny_addresses_stx.push(StacksAddress::try_from(address.as_str()).map_err(
+                |e| SBTCError::InvalidConfig(format!("Failed to parse stacks address: {}", e)),
+            )?);
+        }
```

Comment:
Nit: In general I prefer to avoid mutating vectors as much as possible. Here you seem to just be constructing the vectors from iterators, so I'd instead recommend doing something like:

```
let auto_deny_addresses_btc: Result<Vec<_>, _> = config.auto_deny_addresses_btc.into_iter().flatten().map(String::as_str).map(BitcoinAddress::from_str).collect();
```